StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFormFactory/PaymentSheetFormFactory.swift

```diff
@@ -23,15 +23,27 @@ class PaymentSheetFormFactory {
         case missingV1FromSelectorSpec
     }
 
-    enum SaveMode {
-        /// We can't save the PaymentMethod. e.g., Payment mode without a customer
-        case none
-        /// The customer chooses whether or not to save the PaymentMethod. e.g., Payment mode
-        case userSelectable
-        /// `setup_future_usage` is set on the PaymentIntent or Setup mode
-        case merchantRequired
-    }
-    let saveMode: SaveMode
+    struct SaveMetadata {
+        enum IntentType {
+            case payment_intent
+            case payment_intent_sfu
+            case setup_intent
+        }
+        let intentType: IntentType
+        let hasCustomer: Bool
+        let paymentMethodType: PaymentSheet.PaymentMethodType
```

Comment:
Feels like we can get rid of this and instead calculate the two boolean vars - `isSaving` and `savingIsSelectable` store them on the FormFactory in the initializer.

Reply:
Hmm, we could, but then .deferredIntent logic alongside intentfirst logic.  For example, savingIsSelectable becomes:
```
        if case .paymentIntent(_, let paymentIntent) = intent,
           paymentIntent.setupFutureUsage == .none {
            return hasCustomer && paymentMethod.supportsSaveForFutureUseCheckbox()
        } else if case .deferredIntent(_, let intentConfig) = intent,
                  case .payment(_, _, let setupFutureUsage, _) = intentConfig.mode,
                  setupFutureUsage != .none {
            return hasCustomer && paymentMethod.supportsSaveForFutureUseCheckbox()
        }
```
versus the proposed:
```
            intentType == .payment_intent && hasCustomer && paymentMethodType.supportsSaveForFutureUseCheckbox()
```

I think this struct is an attempt to simplify some of the inputs we use towards showing mandates/showing checkboxes. Eventually it will have a config where merchants can control whether or not they want to show/hide the checkbox as well as whether or not they want the on/off session if not specified. Open to other ideas though.

Reply:
How about this:
```
        var intentType: IntentType
        switch intent {
        case let .paymentIntent(_, paymentIntent):
            saveMode = saveModeFor(merchantRequiresSave: paymentIntent.setupFutureUsage != .none)
            intentType = paymentIntent.setupFutureUsage != .none ? .payment_intent_sfu : .payment_intent
        case .setupIntent:
            saveMode = .merchantRequired
            intentType = .setup_intent
        case .deferredIntent(_, let intentConfig):
            switch intentConfig.mode {
            case .payment(_, _, let setupFutureUsage, _):
                saveMode = saveModeFor(merchantRequiresSave: setupFutureUsage != .none)
                intentType = setupFutureUsage != .none ? .payment_intent_sfu : .payment_intent
            case .setup:
                saveMode = .merchantRequired
                intentType = .setup_intent
            }
        }
  // ... now calculate savingIsSelectable and isSavable
```

Reply:
Found a way to get rid of the metadata struct by using an existing function called isSettingUp on the `Intent` class.